docs/howto/craft-for-platforms/example-python-app.rst

```diff
@@ -0,0 +1,72 @@
+.. _example-python-app:
+
+Example Python app
+==================
+
+This how-to guide covers the steps, decisions, and implementation details that
+are unique when crafting a Python-based snap. We'll work through the aspects
+unique to Python apps by examining an existing recipe.
+
+The process of developing a snap for a Python app builds on top of standard
+Python packaging tools and configuration, making it possible to adapt or
+integrate an app's existing build tooling into the crafting process.
+
+
+Example recipe for liquidctl
+----------------------------
+
+The following code comprises the recipe of a Python project, `liquidctl
+<https://snapcraft.io/liquidctl>`_. This project is both a driver and a CLI
+tool for power and cooling components in PCs.
+
+.. collapse:: liquidctl recipe
+
+  .. code:: yaml
+
+    name: liquidctl
+    summary: a status and control utility to for power, cooling and LED
+    components
+    version: '1.0'
+    description: |
+        liquidctl is a command-line tool to monitor and control the fan speed,
+        LED colour and pump volumes of specific power supplies, motherboards,
+        graphics cards and cooling solutions. The liquidctl snap unofficial and
+        is not endorsed by the upstream project.
+    base: core22
+    confinement: strict
+
+    parts:
+        liquidctl:
+            plugin: python
+            source: .
+            stage-packages:
+                - python3-usb
+
+    apps:
+        liquidctl:
+            command: bin/liquidctl
+            plugs:
+                - raw-usb
+                - hardware-observe
+
+
+Add a part written in Python
+----------------------------
+
+.. code:: yaml
+
+  parts:
+      liquidctl:
+          plugin: python
+          source: .
+          stage-packages:
+              - python3-usb
```

Comment:
If you move the full YAML to its own file (e.g. `example-python-app.yaml`), you can use a literal include for both of these:

```suggestion
  .. literalinclude:: example-python-app.yaml
      :language: yaml

Add a part written in Python
----------------------------
  .. literalinclude:: example-python-app.yaml
      :language: yaml
      :start-at: parts:
      :end-before: apps:
```

Reply:
[342890d](https://github.com/canonical/snapcraft/pull/5160/commits/342890d9e0b2bcb84f35140592fc7c8fe6160b02)